### PR TITLE
Add rename PPL function

### PR DESCRIPTION
**Pull request by kt-eliatra**
### Description
This PR implements the rename PPL function.

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/509

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
> There is one problem, when the rename command is not followed by the fields command, the results contain duplicate values - from both the original and the renamed column, e.g https://github.com/opensearch-project/opensearch-spark/pull/618/files#diff-cb628f2b7ed981862c709fe9b25149c53b5d7eb2fe87a8bb1bbfce4c702352b1R105-R108. I'm not sure how to solve this. Any ideas? Or maybe this can be accepted as a known limitation?

please check out the next code that should help remove the double projected fields:
- [Context projected fields list](https://github.com/opensearch-project/opensearch-spark/blob/88ad15fd208b78c50b061226eb45e9d15f998ce3/ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystPlanContext.java#L73)
- [Usage of this list](https://github.com/opensearch-project/opensearch-spark/blob/88ad15fd208b78c50b061226eb45e9d15f998ce3/ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/utils/ParseStrategy.java#L43)

**Comment by YANG-DB** [comment]
please also add README examples of this command usage

**Comment by YANG-DB** [comment]
@kt-eliatra thanks - this looks like an issue in the `parse` command ?
in addition can you plz add README examples of this command usage

**Comment by salyh** [comment]
> @kt-eliatra thanks - this looks like an issue in the `parse` command ? in addition can you plz add README examples of this command usage

We opened this issue regarding the parse command: https://github.com/opensearch-project/opensearch-spark/issues/650 (thx @kt-eliatra)

But our assumption is that replacing a existing column with new computed column with same name is impossible to do in spark.

**Review by YANG-DB — COMMENTED** [review]
Please add usage example in the readme.md doc

**Comment by kt-eliatra** [comment]
> Some CI test are failing - plz review Please add usage example in the readme.md doc

it should be fine now

**Review by YANG-DB — COMMENTED** [review]
Can u plz resolve the latest conflict ?
thanks

**Comment by kt-eliatra** [comment]
> Can u plz resolve the latest conflict ? thanks

done

**Comment by YANG-DB** [comment]
@kt-eliatra sorry agains - plz resolve latest conflict...

**Comment by kt-eliatra** [comment]
> @kt-eliatra sorry agains - plz resolve latest conflict...

np, done :)

**Comment by LantaoJin** [comment]
OH, there is a PR for `rename`, I opened a duplicated issue https://github.com/opensearch-project/opensearch-spark/issues/697. I am starting to working on the implementation, but I am okey you have already started this. I will stop mine.
However, we should understand the `rename` is not just an alias of field, make sure the source field never be accessed again.
For example. `source=t | rename a as b | fields a` should throw exception. And any other access the `a` field after renaming can't be accessed again.

**Comment by YANG-DB** [comment]
@kt-eliatra plz fix `scalafmtCheck` checkstyle

**Comment by salyh** [comment]
> @kt-eliatra plz fix `scalafmtCheck` checkstyle

done

File: integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLRenameITSuite.scala [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    // Stop all streaming jobs if any
+    spark.streams.active.foreach { job =>
+      job.stop()
+      job.awaitTermination()
+    }
+  }
+
+  test("test single renamed field in fields command") {
+    val frame = sql(s"""
+         | source = $testTable | rename age as renamed_age | fields name, renamed_age
```
Comment (LantaoJin) [inline integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLRenameITSuite.scala:43]: Could you test the following case:
```
| source = $testTable | rename age as renamed_age | fields age
```
Ref https://github.com/opensearch-project/opensearch-spark/pull/618#issuecomment-2378759054

Comment (salyh) [reply integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLRenameITSuite.scala:43]: @lukasz-soszynski-eliatra is on it ...

Comment (lukasz-soszynski-eliatra) [reply integ-test/src/integration/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLRenameITSuite.scala:66]: So far, the query  `source = $testTable | rename age as renamed_age | fields age` has worked "correctly"; that is, the field `age` was present in the query result. I introduced corrections and updated the tests. Currently, the execution of the previously mentioned query causes an error.

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**